### PR TITLE
Better Ghostery Search integration

### PR DESCRIPTION
**Pull request by chrmod**
Send user account info with all Ghostery Search requests so UI can reflect the state of account.

Use templates for tiles.

### CODE REVIEW

File: src/content/top-sites.js [hunk 1]
```diff
@@ -1,61 +1,26 @@
...
-      $title.style.textOverflow = 'clip';
-      $site.appendChild($title);
-
-      $topSitesWrapper.appendChild($site);
+    topSites.slice(0, 5).forEach(site => {
+      const $tile = $tileTemplate.content.cloneNode(true);
+      $tile.querySelector('a').setAttribute('href', site.url);
+      $tile.querySelector('img').setAttribute('src', site.favicon);
+      $tile.querySelector('span').innerText = site.title;
+      $topsites.appendChild($tile);
```
Comment (remusao) [inline src/content/top-sites.js:22]: Is this change related?

File: src/config.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-let DEBUG = false;
+let DEBUG = true;
```
Comment (remusao) [inline src/config.js]: Should this be `false`?
```suggestion
let DEBUG = false;
```

File: src/background.js [hunk 3]
```diff
@@ -100,17 +113,36 @@ async function start() {
       requestHeaders,
     };
   }, { urls: [`${SERP_BASE_URL}/search*`]}, ["blocking", "requestHeaders"]);
+
+  browser.webRequest.onBeforeSendHeaders.addListener(async (details) => {
```
Comment (remusao) [inline src/background.js]: Should this really be `async`?
```suggestion
  browser.webRequest.onBeforeSendHeaders.addListener((details) => {
```

File: src/background.js [hunk 4]
```diff
@@ -15,6 +15,19 @@ class AccessToken {
...
 
+  static parse() {
+    if (!AccessToken.TOKEN) {
+      return {};
+    }
+    const base64Url = AccessToken.TOKEN.split('.')[1];
+    const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+    const jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c) {
+      return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+    }).join(''));
```
Comment (remusao) [inline src/background.js:26]: Do you maybe have an example of token? I'm not sure what the format is.

**Review by remusao — APPROVED** [review]

File: src/background.js (hunk 4 above)
Comment (chrmod) [reply src/background.js:26]: this is JWT starndard

File: src/background.js (hunk 3 above)
Comment (chrmod) [reply src/background.js]: 👍🏻

File: src/config.js (hunk 2 above)
Comment (chrmod) [reply src/config.js]: 👍🏻

File: src/content/top-sites.js (hunk 1 above)
Comment (chrmod) [reply src/content/top-sites.js:22]: it is - instead of creating speed dials HTML by hand, now we use template from the website, so it is ghosterysearch.com that controls the design.
